project-koku/koku
# Ensure there is a 1-1 relationship with sources and providers

`api.provider.models.Sources.MultipleObjectsReturned: get() returned more than one Sources -- it returned 2!` has been seen when adding multiple sources with identical authentication and billing_source information.

Adding unique constraint to koku_uuid field on sources table for extra insurance.

**Testing**
1.  Add an AWS source.  Add a second AWS source with the same authenticaion and billing_source but different name.
2.  Verify platform sources avaiability status information and make sure that proper error message is set for source_id 2
3.  Delete Source 1 and update billing_source to trigger the provider creation.  Verify it's successful.

**Test Results**
[multiple_sources_one_provider_ut.txt](https://github.com/project-koku/koku/files/3870584/multiple_sources_one_provider_ut.txt)
